pkg/build/strategies/sti/postexecutorstep.go

```diff
@@ -114,7 +114,19 @@ func (step *commitImageStep) execute(ctx *postExecutorStepContext) error {
 
 	ctx.labels = createLabelsForResultingImage(step.builder, step.docker, step.image)
 
-	ctx.imageID, err = commitContainer(step.docker, ctx.containerID, cmd, user, step.builder.config.Tag, step.builder.env, ctx.labels)
+	// Set the image entrypoint back to its original value on commit, the running
+	// container has "env" as its entrypoint and we don't want to commit that.
+	entrypoint, err := step.docker.GetImageEntrypoint(step.image)
+	// If the image has no explicit entrypoint, set it to an empty array
+	// so we don't default to leaving the entrypoint as "env" upon commit.
+	if entrypoint == nil {
+		entrypoint = []string{}
+	}
+	if err != nil {
+		return fmt.Errorf("Couldn't get entrypoint of %q image: %v", step.image, err)
+	}
+
+	ctx.imageID, err = commitContainer(step.docker, ctx.containerID, cmd, user, step.builder.config.Tag, step.builder.env, entrypoint, ctx.labels)
```

Comment:
"see an about"?

On Jul 15, 2016 5:16 PM, "Vyacheslav Semushin" <notifications@github.com>
wrote:

> In pkg/build/strategies/sti/postexecutorstep.go
> <https://github.com/openshift/source-to-image/pull/543#discussion_r71041988>
> :
>
> > @@ -114,7 +114,19 @@ func (step *commitImageStep) execute(ctx *postExecutorStepContext) error {
> >
> >  	ctx.labels = createLabelsForResultingImage(step.builder, step.docker, step.image)
> >
> > -	ctx.imageID, err = commitContainer(step.docker, ctx.containerID, cmd, user, step.builder.config.Tag, step.builder.env, ctx.labels)
> > +	// Set the image entrypoint back to its original value on commit, the running
> > +	// container has "env" as its entrypoint and we don't want to commit that.
> > +	entrypoint, err := step.docker.GetImageEntrypoint(step.image)
> > +	// If the image has no explicit entrypoint, set it to an empty array
> > +	// so we don't default to leaving the entrypoint as "env" upon commit.
> > +	if entrypoint == nil {
> > +		entrypoint = []string{}
> > +	}
> > +	if err != nil {
> > +		return fmt.Errorf("Couldn't get entrypoint of %q image: %v", step.image, err)
> > +	}
> > +
> > +	ctx.imageID, err = commitContainer(step.docker, ctx.containerID, cmd, user, step.builder.config.Tag, step.builder.env, entrypoint, ctx.labels)
>
> It was a wish to see an about 5 lines in the the existing unit test ;-)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/source-to-image/pull/543/files/3ba3ee2654fdbff83ee86ee435cb85bcc3be96f0#r71041988>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEvl3gt4h-lwKTBrLqxwyOrLUnL9dsVKks5qV_iMgaJpZM4JNuCH>
> .
>